### PR TITLE
Create scripts for generating a better prompt context

### PR DESCRIPTION
**Pull request by wwwillchen**
Generates a better prompt context by transforming the built (HTML) docs rather than the raw docs.

### CODE REVIEW

File: ai/prompt_context/html_docs_context.txt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+../../site/components/table/index.html
+../../site/components/divider/index.html
+../../site/components/text/index.html
+../../site/components/select/index.html
+../../site/components/icon/index.html
+../../site/components/badge/index.html
+../../site/getting-started/installing/index.html
+../../site/getting-started/quickstart/index.html
+../../site/getting-started/core-concepts/index.html
+../../site/theme/partials/search.html
```
Comment (richard-to) [inline ai/prompt_context/html_docs_context.txt:71]: What is are these partials page?

File: ai/prompt_context/html_docs_context.txt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+../../site/theme/partials/header.html
+../../site/guides/labs/index.html
+../../site/guides/auth/index.html
+../../site/guides/debugging/index.html
+../../site/guides/interactivity/index.html
+../../site/guides/testing/index.html
```
Comment (richard-to) [inline ai/prompt_context/html_docs_context.txt:77]: Similar to a previous comment, but if this is for the ai gen stuff (and not docs bot), then may not need info about testing and stuff.

File: ai/prompt_context/transform_txt.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import os
+import time
+
+import google.generativeai as genai
+
+# Note: this prompt does not work well for every file
+# in particular, it omits too much information from the style API.
+# I use this to cleanup the style doc:
+# "Clean up the formatting for the following Python docs to make it easier for an LLM to understand:"
```
Comment (richard-to) [inline ai/prompt_context/transform_txt.py:17]: So need to manually run code to transform the style doc?

**Review by richard-to — APPROVED** [review]
LGTM - I think you may be able to skip the transform HTML step. I can post a link to some code I have for adding the code snippets and API definitions. Granted, not sure it would be much of an improvement in terms of complexity and final summarized/condensed output.

Also one general question I had was whether this code was for DocBot or for the code gen stuff. Seems like for code gen?

File: ai/prompt_context/html_docs_context.txt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# ../../site/internal/hot-reload/index.html
+# ../../site/internal/toolchain/index.html
+# ../../site/blog/index.html
+# ../../site/blog/2024/05/13/why-mesop/index.html
+# ../../site/blog/2024/07/12/is-mesop--web-components-the-cure-to-front-end-fatigue/index.html
+# ../../site/blog/2024/01/12/visual-editor/index.html
+# ../../site/blog/2023/12/25/hello-mesop/index.html
+# ../../site/blog/archive/2024/index.html
+# ../../site/blog/archive/2023/index.html
+../../site/web-components/index.html
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply ai/prompt_context/html_docs_context.txt:31]: actually, I re-generated this file and I forgot to comment out these files and some others that you pointed out - agree we should leave these out.

File: ai/prompt_context/html_docs_context.txt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+../../site/components/textarea/index.html
+../../site/components/button/index.html
+../../site/components/link/index.html
+../../site/components/table/index.html
+../../site/components/divider/index.html
+../../site/components/text/index.html
+../../site/components/select/index.html
+../../site/components/icon/index.html
+../../site/components/badge/index.html
+../../site/getting-started/installing/index.html
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply ai/prompt_context/html_docs_context.txt:68]: good point - this is just for generating mesop app, and not for docbot.

File: ai/prompt_context/html_docs_context.txt (hunk 1 above)
Comment (wwwillchen) [reply ai/prompt_context/html_docs_context.txt:71]: it's for customizing the docs ui - i've commented them out.

File: ai/prompt_context/transform_txt.py (hunk 3 above)
Comment (wwwillchen) [reply ai/prompt_context/transform_txt.py:17]: yup that's right. I just plugged it into AI studio and copy-pasted, which isn't ideal, but I think we can rework this prompt context pipeline.

**Comment by wwwillchen** [comment]
> LGTM - I think you may be able to skip the transform HTML step. I can post a link to some code I have for adding the code snippets and API definitions. Granted, not sure it would be much of an improvement in terms of complexity and final summarized/condensed output.
> 
> Also one general question I had was whether this code was for DocBot or for the code gen stuff. Seems like for code gen?

yeah please share the code for doing the transformations. the transform HTML step isn't great because the output is quite ugly.

**Pull request merged**